### ISSUE TITLE
Supply optional default naming for authorization

### ISSUE THREAD

**Issue by adammontlake** [Azure/terraform-azurerm-avm-res-network-expressroutecircuit#32]
https://github.com/Azure/terraform-azurerm-avm-res-network-expressroutecircuit/blob/ff54787e667786556d43bb7ca69dd8dedf37add7/main.tf#L5C3-L5C30

**Comment by adammontlake** [comment]
The design as it is, dosen't really allow for automatic naming.
I suggest changing the authorization to the following:
```yaml
variable "express_route_circuit_authorizations" {
  type = map(object({
    name = optional(string, null)
    target_gateway = optional(string, null)
  }))
```
Then we validate that either name OR target has been assigned, if it's name we use as is, if it's target_gateway we put it in a naming convention: auth-express_route_name-target_gateway

@jchancellor-ms @khushal08 - WDYT?

**Comment by adammontlake** [comment]
will be solved as part of pr #29
